shell/platform/darwin/macos/framework/Source/FlutterThreadSynchronizer.mm

```diff
@@ -0,0 +1,132 @@
+#import "flutter/shell/platform/darwin/macos/framework/Source/FlutterThreadSynchronizer.h"
+
+#import <QuartzCore/QuartzCore.h>
+#include <mutex>
+
+@interface FlutterThreadSynchronizer () {
+  std::mutex _mutex;
+  BOOL _shuttingDown;
+  CGSize _contentSize;
+  std::vector<dispatch_block_t> _scheduledBlocks;
+
+  BOOL _beginResizeWaiting;
+
+  // Used to block [beginResize:].
+  std::condition_variable _condBlockBeginResize;
```

Comment:
The mutex is used to synchronize access to `_scheduledBlocks` and `_contentSize`. I don't think this instance can be replaced with `AutoResetWaitableEvent`. The original code had actually too condition variables (one for blocking platform thread and one for blocking raster thread, since we need to do both). The refactor simplified this significantly replacing one of them with event, but I don't think we can safely get rid of both of them without introducing subtle synchronisation issues.

Reply:
Ugh yeah, too much code review in one day for me, apparently. Agreed between the need for wait and for memory fences for synchronisation we're probably stuck with it here.